src/Features/Core/Portable/Completion/AbstractCompletionService.cs

```diff
@@ -277,12 +278,29 @@ private static CompletionItem Disambiguate(CompletionItem item, CompletionItem e
             CancellationToken cancellationToken)
         {
             var context = new CompletionListContext(document, position, triggerInfo, options, cancellationToken);
-
-            await provider.ProduceCompletionListAsync(context).ConfigureAwait(false);
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Why not wrap the entire method body in try-catch-filter? That's what's done in many other places.